Add support to request Mapnik metrics (grid, image)

Algunenano opened the pull request:
Also requesting new release so it can be used in Windshaft (will need tag and npm push)

Algunenano committed:
Enable metrics for grid and images

Algunenano committed:
Release 0.6.18-cdb5

rochoa commented:
Can you add some basic testing?

Algunenano commented:
Added some tests for getGrid, getTile and changed the way the mapnik_metrics is setup (now parsed through normalizeURI instead of directly on open()).

Algunenano commented:
@rochoa Can you please check if this is now OK to be merged?

rochoa commented on lib/mapnik_backend.js:
```diff
@@ -309,7 +313,8 @@ MapnikSource.prototype._createMetatileCache = function(options) {
             x: +coords[2],
-            y: +coords[3]
+            y: +coords[3],
+            metrics: source._mapnik_metrics
```
I've accepted this is the pattern to follow here. It's a bit weird for me because if you think about it, you are passing an options object with references to the same instance object: `source._renderMetatile` can access `_uri.query.limits`, but fair enough.

Algunenano replied:
I'll change it to `source._uri.query.metrics` to at least follow a similar pattern

rochoa commented on lib/mapnik_backend.js:
```diff
@@ -108,6 +108,9 @@ MapnikSource.prototype._normalizeURI = function(uri) {
     uri.query.metatileCache.deleteOnHit = uri.query.metatileCache.hasOwnProperty('deleteOnHit') ?
         as_bool(uri.query.metatileCache.deleteOnHit) : false;
+
+    if (typeof uri.query.mapnik_metrics === "undefined") uri.query.mapnik_metrics = false;
```
Isn't `mapnik_` a bit redundant?

Algunenano replied:
Yup. This comes from a time I didn't have a lot of knowledge about the whole stack. I'll change it to just `metrics`.

rochoa commented on test/error.test.js:
```diff
@@ -43,7 +43,13 @@ describe('Handling Errors ', function() {
               assert.ok(!err, err);
-              done();
+              if (source) {
+                  source.close(function(err) {
```
Should the `close` callback call the `done` function?

Algunenano replied:
I copied & pasted from some example above I think, but yeah it can be changed to source.close(done).

Algunenano committed:
Test: Avoid waiting for sources to automatically close

Algunenano committed:
NPM: Update mime to fix regex DOS issue

Algunenano commented:
Updated to address comments (changing `mapnik_metrics` into just `metrics` and simplifying some close callbacks. Also I noticed that the used `mime` dependency had some vulnerabilities reported so I've updated to the last 1.x which should be ok.
